### PR TITLE
feat: update filterable dropdown spec

### PR DESCRIPTION
**Pull request by skdud4659**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci` ONLY)
- [ ] Not that difficult

### Type of Change
- [ ] New feature
- [ ] Bug fixes
- [x] Feature improvement
- [x] Refactor
- [ ] Others (performance improvement, CI/CD, etc.)

### Affects to
- [ ] Packages
  - [ ] core-lib
  - [x] mirinae
  - [ ] etc 

- [ ] Apps
  - [ ] storybook
  - [ ] web

### Checklist
- Did you check the `lint` and `type`?
- Did you document the changes?
  - Changes in `mirinae` should be reflected in `storybook`.
- Did you test the app after the package changes?

### Description
- New Spec

1. styleType
image

2. selectionLabel
image
image

3. selectionHighlight
image

4. showAlertDot
image

5. showDeleteAllButton
image

### Things to Talk About

### CODE REVIEW

File: packages/mirinae/src/inputs/dropdown/filterable-dropdown/type.ts [hunk 1]
```diff
@@ -1,22 +1,19 @@
 import type { MenuItem } from '@/inputs/context-menu/type';
 
-export type FilterableDropdownMenuItem = MenuItem;
+export type DropdownMenuItem = MenuItem;
```
Comment (WANZARGEN) [inline packages/mirinae/src/inputs/dropdown/filterable-dropdown/type.ts:3]: question: I think console app imports this... isn't it?

File: packages/mirinae/src/inputs/dropdown/filterable-dropdown/story-helper.ts [hunk 2]
```diff
@@ -60,28 +119,82 @@ const extraArgTypes: ArgTypes = {
...
+                summary: false,
             },
         },
         control: {
             type: 'boolean',
         },
     },
-    disabled: {
-        name: 'disabled',
+    showAlertDot: {
```
Comment (WANZARGEN) [inline packages/mirinae/src/inputs/dropdown/filterable-dropdown/story-helper.ts:158]: praise: I like this prop name

File: packages/mirinae/src/inputs/dropdown/filterable-dropdown/PFilterableDropdown.vue [hunk 3]
```diff
@@ -1,117 +1,140 @@
...
-    proxySelected.value = selected;
-    emit('update:selected', selected);
-};
-watch(() => props.selected, (selected) => {
-    if (proxySelected.value !== selected && !isEqual(proxySelected.value, selected)) {
-        updateSelected(selected);
-    }
+const state = reactive({
+    proxyVisibleMenu: useProxyValue<boolean>('visibleMenu', props, emit),
+    proxySelectedItem: useProxyValue<DropdownMenuItem[]>('selected', props, emit),
```
Comment (WANZARGEN) [inline packages/mirinae/src/inputs/dropdown/filterable-dropdown/PFilterableDropdown.vue:94]: praise: explicit name. I like it.

File: packages/mirinae/src/inputs/dropdown/filterable-dropdown/PFilterableDropdown.vue [hunk 4]
```diff
@@ -152,105 +186,95 @@ const handleClearSelection = () => {
...
+        handleUpdateSelected([]);
     }
 };
-const handleTagDelete = (item: FilterableDropdownMenuItem, idx: number) => {
-    const selectedClone = [...proxySelected.value];
+const handleTagDelete = (item: DropdownMenuItem, idx: number) => {
+    const selectedClone = [...state.proxySelectedItem];
     selectedClone.splice(idx, 1);
-    updateSelected(selectedClone);
+    handleUpdateSelected(selectedClone);
```
Comment (WANZARGEN) [inline packages/mirinae/src/inputs/dropdown/filterable-dropdown/PFilterableDropdown.vue:202]: suggestion: We discussed it and decided not to call event handler but just bind it at event.
I think previous name is better in this case.

**Review by WANZARGEN — APPROVED** [review]
LGTM! I left a very nitpick comment. Thank you for your effort!

File: packages/mirinae/src/inputs/dropdown/filterable-dropdown/type.ts (hunk 1 above)
Comment (skdud4659) [reply packages/mirinae/src/inputs/dropdown/filterable-dropdown/type.ts:3]: I'll check! thx!

**Review by sulmoJ — APPROVED** [review]
LGTM~

**Review by yuda110 — APPROVED** [review]
LGTM!!! 👍

**Pull request merged**